hibernate-core/src/test/java/org/hibernate/test/dialect/functional/SQLServerDialectTempTableCollationTest.java

```diff
@@ -0,0 +1,159 @@
+
+	private String originalDBCollation;
+	private final String changedDBCollation = "SQL_Latin1_General_CP437_BIN";
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.setProperty( AvailableSettings.KEYWORD_AUTO_QUOTING_ENABLED, Boolean.TRUE.toString() );
+		return configuration;
+	}
+
+	@AfterClassOnce
+	protected void revertBackOriginalDBCollation() {
+		if ( originalDBCollation != null && !changedDBCollation.equals( originalDBCollation ) ) {
+			BootstrapServiceRegistry bootRegistry = buildBootstrapServiceRegistry();
+			StandardServiceRegistryImpl serviceRegistry = buildServiceRegistry(
+					bootRegistry,
+					constructConfiguration()
+			);
+			try (Connection connection = serviceRegistry.getService( JdbcServices.class )
+					.getBootstrapJdbcConnectionAccess()
+					.obtainConnection();
+				 Statement statement = connection.createStatement()) {
+				connection.setAutoCommit( true );
+				statement.executeUpdate( "ALTER DATABASE CURRENT COLLATE " + originalDBCollation );
+			}
+			catch (SQLException e) {
+				log.debug( e.getMessage() );
+			}
+			finally {
+				serviceRegistry.destroy();
+			}
+		}
+	}
+
+	protected void buildSessionFactory() {
+		BootstrapServiceRegistry bootRegistry = buildBootstrapServiceRegistry();
+		StandardServiceRegistryImpl serviceRegistry = buildServiceRegistry( bootRegistry, constructConfiguration() );
+
+		try {
+			try ( Connection connection = serviceRegistry.getService( JdbcServices.class ).getBootstrapJdbcConnectionAccess().obtainConnection();
+				 Statement statement = connection.createStatement() ) {
+				connection.setAutoCommit( true );
+				try ( ResultSet rs = statement.executeQuery( "SELECT SERVERPROPERTY('collation')" ) ) {
+					rs.next();
+					String instanceCollation = rs.getString( 1 );
+					Assert.assertNotEquals( instanceCollation, changedDBCollation );
+				}
+			}
+			catch (SQLException e) {
+				log.debug( e.getMessage() );
+			}
+			try ( Connection connection = serviceRegistry.getService( JdbcServices.class ).getBootstrapJdbcConnectionAccess().obtainConnection();
+				  Statement statement = connection.createStatement() ) {
+				connection.setAutoCommit( true );
+				try ( ResultSet rs = statement.executeQuery( "SELECT CONVERT (varchar(256), DATABASEPROPERTYEX(DB_NAME(),'collation'))" ) ) {
+					rs.next();
+					originalDBCollation = rs.getString( 1 );
+				}
+			}
+			catch (SQLException e) {
+				log.debug( e.getMessage() );
+			}
+			try ( Connection connection = serviceRegistry.getService( JdbcServices.class ).getBootstrapJdbcConnectionAccess().obtainConnection();
+				 Statement statement = connection.createStatement() ) {
+				connection.setAutoCommit( true );
+				statement.executeUpdate( "ALTER DATABASE CURRENT COLLATE " + changedDBCollation );
+			}
+			catch (SQLException e) {
+				log.debug( e.getMessage() );
```

Comment:
Instead of logging a debug message, I think it would be better to have a catch clause in the outer try-finally block that rethrows as RuntimeException. Or is there a valid use case for a SQLException being thrown?